### ISSUE TITLE
Cannot connect to coordination server warning

### ISSUE THREAD

**Issue by ARizzo35** [tailscale/tailscale#13906]
### What is the issue?

I have noticed after updating the Tailscale client to 1.76, machines on my tailnet are showing warnings about connecting to the coordination server more frequently. On my MacOSX laptop, basically every time I change locations/connect to a different router, I get a warning in the Tailscale client about not being able to connect to the coordination server:

```
# Health check:
#     - Unable to connect to the Tailscale coordination server to synchronize the state of your tailnet. Peer reachability might degrade over time.
```

Stopping and starting the Tailscale client resolves the issue, but I would expect the client to be able to resolve this on its own without intervention. Did something change with how the Tailscale client retries connection to the coordination server?

### Steps to reproduce

- Connect to Tailscale coordination server
- Change local internet connection
- 

### Are there any recent changes that introduced the issue?

Seems related to client update to 1.76 but unsure

### OS

Linux, macOS

### Tailscale version

1.76.1

**Comment by bradfitz** [comment]
Got a `tailscale bugreport` for us to look at logs?

Any other VPN/security software running your machines? Who's your ISP?

**Comment by ARizzo35** [comment]
We're using headscale so logging is disabled (`BUG-NO-LOGS-NO-SUPPORT-this-node-has-had-its-logging-disabled`).

No other VPN/security software running. Google Fiber ISP for the most part.

**Comment by ARizzo35** [comment]
This warning comes up pretty frequently now if there's anything I can try before restarting Tailscale to help debug.

**Comment by glidingthrough** [comment]
Similar thing here, running 1.76.6 on linux.

```
$ tailscale bugreport
BUG-97784dae3330eb0a6231c02960407625ceea5a4595ea75b42fdac96632681bc8-20241107090731Z-2e64f6866cb4a2f7

```

**Comment by ShaneSkyWalker** [comment]
Same issue.
`root@ImmortalWrt:~# tailscale bugreport
BUG-4147a45c67caeab654a759768c0350dc936a20c9ed4c9e19b269cd7319eaeb7e-20241108033051Z-2763831338ea8004`

**Comment by hallelujah-shih** [comment]
same issue.

system: Fedora41

**Comment by itfsw** [comment]
same issue
system: centos7
tailscale: 1.76.1

**Comment by kbsec** [comment]
ditto: 
Linux git 6.6.51+rpt-rpi-2712 #1 SMP PREEMPT Debian 1:6.6.51-1+rpt3 (2024-10-08) aarch64 GNU/Linux
1.76.6
  tailscale commit: 1edcf9d466ceafedd2816db1a24d5ba4b0b18a5b
  other commit: d0a6cd8b27eb46f6dec31425499159f7949be7f9
  go version: go1.23.1

issue is fixed when restarting tailscaled but  this is required eveyt ime the coordination server changes.

**Comment by Darkhand81** [comment]
Same here:

`BUG-ecc1c092a12cb277458086c496773d485fb156927bf3f662a42fb8888edb9e41-20241201034623Z-d88f3f5a022ce815`

Connects at first for a few minutes, then bounces between direct and relay, then fails. Once things fail netcheck cannot find my external IP anymore:

Immediately after starting tailscale:

```
tailscale netcheck

Report:
        * UDP: true
        * IPv4: yes, xx.x.xxx.xxx:58945
        * IPv6: no, but OS has support
        * MappingVariesByDestIP: false
        * PortMapping:
        * Nearest DERP: Chicago
        * DERP latency:
                - ord: 17.5ms  (Chicago)
                - tor: 27ms    (Toronto)
                - nyc: 33.5ms  (New York City)
                - den: 40.6ms  (Denver)
                - dfw: 44.8ms  (Dallas)
                - .......
```

After a few minutes:

```
tailscale netcheck

Report:
        * UDP: false
        * IPv4: (no addr found)
        * IPv6: no, but OS has support
        * MappingVariesByDestIP:
        * PortMapping:
        * CaptivePortal: false
        * Nearest DERP: Chicago
        * DERP latency:
                - ord: 20.2ms  (Chicago)
                - tor: 25.7ms  (Toronto)
                - nyc: 34ms    (New York City)
                - den: 44.3ms  (Denver)
                - dfw: 46.9ms  (Dallas)
                - .......
```

**Comment by xylobol** [comment]
Seeing this too. openSUSE Tumbleweed, using the packaged Tailscale. 

```
xylobol@anastasia:~> tailscale version
1.76.6
  tailscale commit: 1edcf9d466ceafedd2816db1a24d5ba4b0b18a5b-dirty
  go version: go1.23.3
```

Seems to happen a lot after the Headscale server restarts.

**Comment by xylobol** [comment]
This seems to be impacting the Android client as well: https://www.reddit.com/r/Tailscale/comments/1gahenk/tailscale_android_out_of_sync/

I rolled back to 1.74 on my laptop (`anastasia`), and the issue persisted. When I'm off work, I'll test with a few more veersions.

**Comment by xylobol** [comment]
I switched to the official Tailscale repos, and downgraded to versions 1.60.1 and 1.40. The same error (different log text, but trying to post to the same URL and timing out) was present in logs, and `tailscale status` showed
```
# Health check:
#     - not in map poll
````

In all cases, restarting `tailscaled` is a workaround. 

I've been testing by restarting the Headscale service on the machine that runs it. My Linux and Android devices don't respond well to this, meanwhile the Windows machines handle it with no problem.

**Comment by ARizzo35** [comment]
For anyone else still having this issue, I was able to root cause this from the following log:

`control: controlhttp: forcing port 443 dial due to recent noise dial`

It seems like the client can occasionally force HTTPS instead of HTTP even if the coordination server does not support it. Adding HTTPS support to the coordination server solved the issue for me.

**Comment by temporal-dude** [comment]
Why did you close the issue, seems like an active bug.  Is this a won't fix situation?

**Comment by ARizzo35** [comment]
> Why did you close the issue, seems like an active bug. Is this a won't fix situation?

My issue is resolved after doing what I posted above. If you're still having issues, it's unrelated to the problem I was having. Feel free to open another issue.

**Comment by Neobond** [comment]
I had this issue yesterday and I have been able to resolve it by following this guide https://tailscale.com/kb/1082/firewall-ports

Specifically allowing LAN devices to connect from ports 80 and 443, plus:

![Image](https://github.com/user-attachments/assets/c7a74233-4ca0-41ed-a9a3-6db7e8a668ac)

The error came back briefly but I restarted the Tailscale package on my NAS (DSM 7.2.2) and Android, and it has been fine since.

**Comment by Abdelkrim** [comment]
Unable to connect to the Tailscale coordination server to synchronize the state of your tailnet. Peer reachability might degrade over time.

+ environment: WSL2 on windows 11 Pro
+ I just restarted the machine a its 'sleep mode'

BUG-eeec141a05acf04d8e299171b54d95c0c7e6d90668b6ba3975cc6ed23fd7764c-20260222230852Z-55efc1e686a16a46


tailscale netcheck
2026/02/23 00:11:00 portmap: monitor: gateway and self IP changed: gw=172.23.208.1 self=172.23.210.156

Report:
        * Time: 2026-02-22T23:11:01.197668102Z
        * UDP: true
        * IPv4: yes, X.X.X.X:61233
        * IPv6: no, but OS has support
        * MappingVariesByDestIP: false
        * PortMapping: 
        * Nearest DERP: Frankfurt
        * DERP latency:
                - fra: 12.6ms  (Frankfurt)
                - ams: 19ms    (Amsterdam)
                - nue: 20.5ms  (Nuremberg)
                - lhr: 21.7ms  (London)
                - par: 24.4ms  (Paris)
                - waw: 37ms    (Warsaw)
                - hel: 41.5ms  (Helsinki)
                - mad: 51.6ms  (Madrid)
                - iad: 100.4ms (Ashburn)
                - tor: 100.5ms (Toronto)
                - nyc: 100.6ms (New York City)
                - ord: 110.7ms (Chicago)
                - mia: 118.2ms (Miami)
                - den: 129.3ms (Denver)
                - dfw: 129.3ms (Dallas)
                - dbi: 134.5ms (Dubai)
                - sea: 151.8ms (Seattle)
                - blr: 154.6ms (Bengaluru)
                - sfo: 159.7ms (San Francisco)
                - sin: 167.4ms (Singapore)
                - nai: 185.1ms (Nairobi)
                - sao: 214.9ms (São Paulo)
                - hkg: 214.9ms (Hong Kong)
                - jnb: 214.9ms (Johannesburg)
                - lax: 255.1ms (Los Angeles)
                - syd:         (Sydney)
                - tok:         (Tokyo)
                - hnl:         (Honolulu)

**Comment by petertulala** [comment]
Same issue. Tailscale coordination server not reachable, probably blocked by the ISP.

Bugreport:`BUG-2638ed1df7c644ac388502a8d7f91f7f88672ee2ac2283df9bf193c691f20ba5-20260325124445Z-1d948f0194c12caa`

**Comment by Neobond** [comment]
> Same issue. Tailscale coordination server not reachable, probably blocked by the ISP.
> 
> Bugreport:`BUG-2638ed1df7c644ac388502a8d7f91f7f88672ee2ac2283df9bf193c691f20ba5-20260325124445Z-1d948f0194c12caa`

One thing that helped me on my phone was to not use tailscale subnets, not had a connection error since yesterday.

**Comment by shayneoneill** [comment]
Anyone actually know what causes this?!

BUG-fd0236569979f2881e91568dcf41531874ea10e186df0ec2d5128770b0e52439-20260414015131Z-99280ab423c1c41e

**Comment by jn0** [comment]
BUG-a14d88be3289e4092c42b71b42d441c9c2e7ed5cff5d725c287f3e4796ed8485-20260728134500Z-d514705e6bd90004
